### ISSUE TITLE
Do we delete discount amount if GERS breaks the order line?

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#630]
_Imported from trac ticket 630.
Created by **kbower**
Opened in trac: Jan 31, 2011, 2:14:08 PM
Last modified in trac: Aug 7, 2014, 10:14:41 AM_
***
If GERS breaks an order line, there is confusion as to exactly what happens. 

If the SO_LN2DISC records are deleted, then the next time we save the order, our system will remove the discount amount, expecting that it be reapplied, which will fail.  The end result would be a deleted discount.

According to JR:
_BUT:  As soon as the lines auto split, ALL the DISCOUNTS disappear and you have to RE-ENTER the discount code on EACH line again..._
Jesús R Capó, VP, CIO

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 31, 2011, 2:14:54 PM:_

GERS written order for qty 2:

```
SQL> select * from orderdetails where orderid = '0128104AOFG';

ORDERID            LINEID PRODUCTID QTYORDERED  ITEMPRICE V VOIDDATE             FI FILLLO COM      SPIFF
-------------- ---------- --------- ---------- ---------- - -------------------- -- ------ --- ----------
ORIGINALINVENTORYCOST FILLDATE             INVENTORYCOSTDATE    INVENTORYRECEIPTDATE ADDEDDATE
--------------------- -------------------- -------------------- -------------------- --------------------
MAINORDER   PACKAGEID SALESPERSO SALESPERSO PERCENTOFSALE1 PERCENTOFSALE2 P   UPCHARGE   DISCOUNT VALUEPROD
----------- --------- ---------- ---------- -------------- -------------- - ---------- ---------- ---------
VALUEPRODUCTLINEID VALUEFORP VALUEFORLINEID DELIVERYPOINTS T S PICKUPDISCOUNT S D
------------------ --------- -------------- -------------- - - -------------- - -
0128104AOFG             1 700144215          2        104 N                                A

0128104AOFG           SLSTRN                           100                Z                 25.99
                                                         1 N N              0 N Y


SQL> select * from SO_LN2DISC where del_doc_num = '0128104AOFG';

DEL_DOC_NUM    DEL_DOC_LN# DIS   DISC_AMT    SEQ_NUM DIS
-------------- ----------- --- ---------- ---------- ---
DISC_CMNT                                                                DISC_PCT
---------------------------------------------------------------------- ----------
0128104AOFG              1 LTR      25.99          1
                                                                               20


SQL> select DISC_CD from so where del_doc_num = '0128104AOFG';

DIS
---
LIN

SQL>
```

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 31, 2011, 8:52:11 PM:_

**After GERS breaks line:**

```
SQL>  select * from orderdetails where orderid = '0128104AOFG';

ORDERID            LINEID PRODUCTID QTYORDERED  ITEMPRICE V VOIDDATE             FI FILLLO COM      SPIFF
-------------- ---------- --------- ---------- ---------- - -------------------- -- ------ --- ----------
ORIGINALINVENTORYCOST FILLDATE             INVENTORYCOSTDATE    INVENTORYRECEIPTDATE ADDEDDATE
--------------------- -------------------- -------------------- -------------------- --------------------
MAINORDER   PACKAGEID SALESPERSO SALESPERSO PERCENTOFSALE1 PERCENTOFSALE2 P   UPCHARGE   DISCOUNT VALUEPROD
----------- --------- ---------- ---------- -------------- -------------- - ---------- ---------- ---------
VALUEPRODUCTLINEID VALUEFORP VALUEFORLINEID DELIVERYPOINTS T S PICKUPDISCOUNT S D
------------------ --------- -------------- -------------- - - -------------- - -
0128104AOFG             1 700144215          1        104 N                      04 ACC    A            0
                   59 31-JAN-2011 00:00:00 30-SEP-2009 00:00:00 19-AUG-2009 00:00:00
0128104AOFG           SLSTRN                           100                F                 25.99
                                                         1 N N              0 N Y

0128104AOFG             2 700144215          1        104 N                                A            0

0128104AOFG           SLSTRN                           100                Z                 25.99
                                                         1 N N              0 N Y


SQL> select * from SO_LN2DISC where del_doc_num = '0128104AOFG';

DEL_DOC_NUM    DEL_DOC_LN# DIS   DISC_AMT    SEQ_NUM DIS
-------------- ----------- --- ---------- ---------- ---
DISC_CMNT                                                                DISC_PCT
---------------------------------------------------------------------- ----------
0128104AOFG              1 LTR      25.99          1
                                                                               20

0128104AOFG              2 LTR      25.99          1
                                                                               20


SQL> select DISC_CD from so where del_doc_num = '0128104AOFG';

DIS
---
LIN
```
**_Therefore, at least the Kane's version of GERS processes this correctly, copying SO_LN2DISC records when breaking lines_**

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 31, 2011, 8:55:55 PM:_

Check eld and ivan smith, because eld says these records are not copied.

**Comment by kbower** [comment]
_**kbower** trac comment on Jun 1, 2013, 11:25:32 AM:_

The answer is, Yes, we do sometimes delete the entire discount, because GERS bug that won't update orderdetaildiscounts primary key when splitting the order.

**Comment by kbower** [comment]
_**kbower** trac status change on Aug 7, 2014, 10:14:40 AM:_ **closed**

**Comment by kbower** [comment]
_**kbower** trac comment on Aug 7, 2014, 10:14:41 AM:_

Closed as duplicate of [#1060](1060).
